pkg/smartcontract/binding/generate.go

```diff
@@ -0,0 +1,255 @@
+		return "map[string]interface{}"
+	case smartcontract.InteropInterfaceType:
+		return "interface{}"
+	case smartcontract.VoidType:
+		return ""
+	default:
+		panic("unreachable")
+	}
+}
+
+func templateFromManifest(cfg Config) (contractTmpl, error) {
+	hStr := ""
+	for _, b := range cfg.Hash.BytesBE() {
+		hStr += fmt.Sprintf("\\x%02x", b)
+	}
+
+	ctr := contractTmpl{
+		PackageName:  cfg.Package,
+		ContractName: cfg.Manifest.Name,
+		Hash:         hStr,
+	}
+	if ctr.PackageName == "" {
+		buf := bytes.NewBuffer(make([]byte, 0, len(cfg.Manifest.Name)))
+		for _, r := range cfg.Manifest.Name {
+			if unicode.IsLetter(r) {
+				buf.WriteRune(unicode.ToLower(r))
+			}
+		}
+
+		ctr.PackageName = buf.String()
+	}
+
+	imports := make(map[string]struct{})
+	seen := make(map[string]bool)
+	for _, m := range cfg.Manifest.ABI.Methods {
+		seen[m.Name] = false
+	}
+	for _, m := range cfg.Manifest.ABI.Methods {
+		if m.Name[0] == '_' {
+			continue
+		}
+
+		imports["github.com/nspcc-dev/neo-go/pkg/interop/contract"] = struct{}{}
+		imports["github.com/nspcc-dev/neo-go/pkg/interop/neogointernal"] = struct{}{}
+
+		// Consider `perform(a)` and `perform(a, b)` methods.
+		// First, try to export the second method with `Perform2` name.
+		// If `perform2` is already in the manifest, use `perform_2` with as many underscores
+		// as needed to eliminate name conflicts. It will produce long names in certain circumstances,
+		// but if the manifest contains lots of similar names with trailing underscores, delicate naming
+		// was probably not the goal.
+		name := m.Name
+		if v, ok := seen[name]; ok && !v {
+			seen[name] = true
+		} else {
+			suffix := strconv.Itoa(len(m.Parameters))
+			for ; seen[name]; name = m.Name + suffix {
+				suffix = "_" + suffix
+			}
+			seen[name] = true
+		}
+
+		mtd := methodTmpl{
+			Name:     upperFirst(name),
+			NameABI:  m.Name,
+			CallFlag: callflag.All.String(),
+			Comment:  fmt.Sprintf("invokes `%s` method of contract.", m.Name),
+		}
+		if f, ok := cfg.CallFlags[m.Name]; ok {
+			mtd.CallFlag = f.String()
+		} else if m.Safe {
+			mtd.CallFlag = callflag.ReadOnly.String()
+		}
+		for i := range m.Parameters {
+			name := m.Parameters[i].Name
+			if name == "" {
+				name = fmt.Sprintf("arg%d", i)
+			}
+
+			var typeStr string
+			if over, ok := cfg.Overrides[m.Name+"."+name]; ok {
+				typeStr = over.TypeName
+				if over.Package != "" {
+					imports[over.Package] = struct{}{}
+				}
+			} else {
+				typeStr = scTypeToGo(m.Parameters[i].Type)
+			}
+
+			mtd.Arguments = append(mtd.Arguments, paramTmpl{
+				Name: name,
+				Type: typeStr,
+			})
+		}
+
+		if over, ok := cfg.Overrides[m.Name]; ok {
+			mtd.ReturnType = over.TypeName
+			if over.Package != "" {
+				imports[over.Package] = struct{}{}
+			}
+		} else {
+			mtd.ReturnType = scTypeToGo(m.ReturnType)
+			switch m.ReturnType {
+			case smartcontract.PublicKeyType, smartcontract.Hash160Type, smartcontract.Hash256Type, smartcontract.SignatureType:
```

Comment:
Add `smartcontract.InteropInterfaceType` to this list?

Reply:
Fixed.